src/instrumentsscene/view/systemobjectslayertreeitem.cpp

```diff
@@ -0,0 +1,260 @@
+/*
+ * SPDX-License-Identifier: GPL-3.0-only
+ * MuseScore-Studio-CLA-applies
+ *
+ * MuseScore Studio
+ * Music Composition & Notation
+ *
+ * Copyright (C) 2024 MuseScore Limited
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+#include "systemobjectslayertreeitem.h"
+
+#include "engraving/dom/timesig.h"
+
+#include "layoutpanelutils.h"
+#include "translation.h"
+#include "log.h"
+
+using namespace mu::instrumentsscene;
+using namespace mu::notation;
+using namespace mu::engraving;
+
+static QString formatLayerTitle(const SystemObjectGroups& groups)
+{
+    if (groups.empty()) {
+        return muse::qtrc("instruments", "System objects");
+    }
+
+    QString title;
+
+    const size_t lastIdx = groups.size() - 1;
+    for (size_t i = 0; i <= lastIdx; ++i) {
+        const SystemObjectsGroup& group = groups.at(i);
+        if (!isSystemObjectsGroupVisible(group)) {
+            continue;
+        }
+
+        if (title.isEmpty()) {
+            title = translatedSystemObjectsGroupName(group);
+        } else {
```

Comment:
```suggestion
        if (title.isEmpty()) {
            title = translatedSystemObjectsGroupName(group);
            continue;
        }
```

^ Flattens the nesting a bit